### PR TITLE
AnubisNetworks Cyberfeed bot revised and added HTTP timeouts

### PR DESCRIPTION
**Pull request by wagner-certat**
* dropped the collector, just use the generic http stream collector (pycurl is no longer needed)
* extended the parser to use more fields
* readded generic http stream collector to docs

### CODE REVIEW

File: intelmq/bots/BOTS [hunk 1]
```diff
@@ -61,6 +61,20 @@
...
+            "module": "intelmq.bots.collectors.http.collector_http_stream",
+            "parameters": {
+                "feed": "",
+                "provider": "",
+                "http_url": "<insert url of feed>",
+                "http_username": null,
+                "http_password": null,
+                "ssl_client_certificate": null,
+                "strip_lines": true,
+                "rate_limit": 3600
```
Comment (wagner-certat) [inline intelmq/bots/BOTS]: Tough this isn't the scope of the PR I now implemented HTTP timeouts for all collector bots using HTTP queries. Also added doc how this specific bot handles timeouts.

Comment (aaronkaplan) [inline intelmq/bots/BOTS]: thx

**Comment by SYNchroACK** [comment]
I'm currently reviewing this one with @pedromreis in order to understand the issues and the possible fixes that he had with his experience with this feed.

File: intelmq/bots/parsers/bitsight/parser.py [hunk 2]
```diff
@@ -1,33 +1,66 @@
 # -*- coding: utf-8 -*-
```
Comment (SYNchroACK) [inline intelmq/bots/parsers/bitsight/parser.py]: change the folder name to `./anubisnetworks/`

File: intelmq/bots/parsers/bitsight/parser.py [hunk 3]
```diff
@@ -1,33 +1,66 @@
...
+                           "region": 'source.geolocation.region',
+                           "city": 'source.geolocation.city',
+                           "asn_name": 'source.as_name',
+                           "asn": 'source.asn',
+                           "longitude": 'source.geolocation.longitude',
+                           "latitude": 'source.geolocation.latitude',
+                           }
+
 
 class BitsightParserBot(Bot):
```
Comment (SYNchroACK) [inline intelmq/bots/parsers/bitsight/parser.py]: replace by `class AnubisNetworksParserBot(Bot):`

File: intelmq/tests/bots/parsers/bitsight/test_parser.py [hunk 4]
```diff
@@ -5,38 +5,37 @@
 import intelmq.lib.test as test
```
Comment (SYNchroACK) [inline intelmq/tests/bots/parsers/bitsight/test_parser.py]: change folder name to `./anubisnetworks/`

File: intelmq/tests/bots/parsers/bitsight/test_parser.py [hunk 5]
```diff
@@ -5,38 +5,37 @@
...
-                  "mVtb3RlX2FkZHIiOiIxNTIuMTY2LjExOS4yIiwicmVtb3RlX3"
-                  "BvcnQiOiI2NTExOCIsInNlcnZlcl9hZGRyIjoiNTIuMTguMTk"
-                  "2LjE2OSIsInNlcnZlcl9wb3J0IjoiOTc5NiJ9LCJfdHMiOjE0"
-                  "NjExMDc3NjgsIl9nZW9fZW52X3JlbW90ZV9hZGRyIjp7ImNvd"
-                  "W50cnlfbmFtZSI6IkRvbWluaWNhbiBSZXB1YmxpYyJ9fQ=="
+                  "raw": EXAMPLE_REPORT['raw'],
+                  'extra': '{"request_method": "POST"}',
                   }
 
 EXAMPLE_REPORT2 = {"feed.name": "BitSight",
```
Comment (SYNchroACK) [inline intelmq/tests/bots/parsers/bitsight/test_parser.py]: replace by AnubisNetworks

File: docs/Feeds.md [hunk 6]
```diff
@@ -182,15 +182,16 @@ Parser: Bitcash Blocklist Feed (`intelmq.bots.parsers.bitcash.parser`)
 Blocklist provided by bitcash.cz of banned IPs for service abuse, this includes scanning, sniffing, harvesting, and dos attacks.
 
 
-## BitSight Ciberfeed Stream
+## BitSight Cyberfeed Stream
```
Comment (aaronkaplan) [inline docs/Feeds.md]: +1

It's officially called: "Cyberfeed" with a "y".

https://www.anubisnetworks.com/products/threat-intelligence/cyberfeed

File: intelmq/bots/collectors/http/collector_http_stream.py [hunk 7]
```diff
@@ -8,12 +22,30 @@
...
-            raise ValueError('Connection Failed.')
+            self.logger.exception('Connection Failed.')
         else:
+            if req.status_code // 100 != 2:
+                raise ValueError('HTTP response status code was {}.'
+                                 ''.format(req.status_code))
+
             for line in req.iter_lines():
                 if self.parameters.strip_lines:
                     line = line.strip()
```
Comment (wagner-certat) [inline intelmq/bots/collectors/http/collector_http_stream.py]: As opposed to the Pedros collector, the HTTP Stream collector uses requests and iterates over all lines using requests' methods and sends lines as individual reports. So your proposal does not apply.

Also, we are using this bot in production and haven't encountered any problems so far.

File: intelmq/bots/BOTS [hunk 8]
```diff
@@ -202,9 +207,9 @@
             "module": "intelmq.bots.parsers.bitcash.parser",
             "parameters": {}
         },
-        "BitSight Ciberfeed Stream": {
-            "description": "Bitsight Ciberfeed parser is the bot reponsible to parse and sanitize the information.",
-            "module": "intelmq.bots.parsers.bitsight.parser",
+        "AnubisNetworks Ciberfeed Stream": {
```
Comment (SYNchroACK) [inline intelmq/bots/BOTS]: after this fix, feel free to merge. Thank you @wagner-certat

File: intelmq/tests/bots/anubisnetworks/test_parser.py [hunk 9]
```diff
@@ -3,43 +3,42 @@
...
+                  "__type": "Event",
+                  "feed.name": "AnubisNetworks",
+                  "raw": EXAMPLE_REPORT['raw'],
+                  'extra': '{"request_method": "POST"}',
                   }
 
-EXAMPLE_REPORT2 = {"feed.name": "BitSight",
+EXAMPLE_REPORT2 = {"feed.name": "AnubisNetworks",
                    "feed.accuracy": 100.0,
                    "feed.url": "http://alerts.bitsighttech.com:8080/stream?",
```
Comment (SYNchroACK) [inline intelmq/tests/bots/anubisnetworks/test_parser.py]: is this correct?

File: docs/Feeds.md (hunk 6 above)
Comment (SYNchroACK) [inline docs/Feeds.md]: replace by 'AnubisNetworks Cyberfeed Stream'

File: intelmq/tests/bots/anubisnetworks/test_parser.py (hunk 9 above)
Comment (wagner-certat) [inline intelmq/tests/bots/anubisnetworks/test_parser.py]: The api key is missing, but it doesn't matter.

**Pull request merged**